tests/api2/test_108_catalog.py

```diff
@@ -0,0 +1,216 @@
+    'deluge',
+    'esphome',
+    'freshrss',
+    'gaps',
+    'grocy',
+    'handbrake',
+    'heimdall',
+    'home-assistant',
+    'jackett',
+    'jellyfin',
+    'kms',
+    'lazylibrarian',
+    'lidarr',
+    'lychee',
+    'navidrome',
+    'node-red',
+    'nzbget',
+    'nzbhydra',
+    'ombi',
+    'organizr',
+    'qbittorrent',
+    'radarr',
+    'readarr',
+    'sabnzbd',
+    'sonarr',
+    'tautulli',
+    'traefik',
+    'transmission',
+    'truecommand',
+    'tvheadend',
+    'unifi',
+    'zwavejs2mqtt'
+]
+
+truechart_catalog = {
+    'label': 'TRUECHARTS',
+    'repository': truechart_repository,
+    'branch': 'master',
+    'builtin': False,
+    'preferred_trains': ['charts'],
+    'location': '/tmp/ix-applications/catalogs/github_com_ericbsd_charts_git_master',
+    'id': 'TRUECHARTS'
+}
+
+
+def test_01_get_catalog_list():
+    results = GET('/catalog/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list), results.text
+
+
+def test_02_search_for_official_catalog_with_the_label():
+    results = GET('/catalog/?label=OFFICIAL')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), list), results.text
+    assert results.json()[0]['id'] == 'OFFICIAL', results.text
+
+
+def test_03_get_official_catalog_with_id():
+    results = GET('/catalog/id/OFFICIAL/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['label'] == 'OFFICIAL', results.text
+
+
+def test_04_verify_official_catalog_repository_with_id():
+    results = GET('/catalog/id/OFFICIAL/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['repository'] == official_repository, results.text
+
+
+def test_05_verify_official_catalog_repository_with_id():
+    results = GET('/catalog/id/OFFICIAL/')
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['repository'] == official_repository, results.text
+
+
+def test_06_validate_official_catalog():
+    results = POST('/catalog/validate/', 'OFFICIAL')
+    assert results.status_code == 200, results.text
+    assert results.json() is None, results.text
+
+
+def test_07_sync_official_catalog():
+    results = POST('/catalog/sync/', 'OFFICIAL')
+    assert results.status_code == 200, results.text
+    assert results.json() is None, results.text
+
+
+@pytest.mark.parametrize('chart', official_chart)
+def test_08_get_official_catalog_item(chart):
+    payload = {
+        'label': 'OFFICIAL',
+        'options': {
+            'cache': True
+        }
+    }
+    results = POST('/catalog/items/', payload)
+    assert results.status_code == 200, results.text
+    assert isinstance(results.json(), dict), results.text
+    assert results.json()['charts'][chart]['name'] == chart
```

Comment:
If `chart` is not present, this will raise a `KeyError`, we should probably do `chart in results.json()['charts']`

Reply:
Totally agreed with that.

Reply:
Done